test/Benchmarks/src/Table/Arithmetic.enso

```diff
@@ -8,45 +8,117 @@ polyglot java import java.lang.Long as Java_Long
 
 options = Bench.options . set_warmup (Bench.phase_conf 3 5) . set_measure (Bench.phase_conf 3 5)
 
-
-create_table : Table
-create_table num_rows =
+create_vectors num_rows =
     x = Vector.new num_rows i->
         i+1
     y = Vector.new num_rows i->
         if i % 10 < 2 then Java_Long.MAX_VALUE else i+1
     u = Vector.new num_rows i->
         10 + (i % 100)
+    z = Vector.new num_rows i->
+        if i % 10 < 2 then Nothing else i+1
 
-    t = Table.new [["X", x], ["Y", y], ["U", u]]
-
-    assert condition =
-        if condition.not then Panic.throw "Assertion failed"
+    [x, y, u, z]
 
-    assert ((t.at "X" . value_type) == Value_Type.Integer)
-    assert ((t.at "Y" . value_type) == Value_Type.Integer)
-    assert ((t.at "U" . value_type) == Value_Type.Integer)
+create_table : Table
+create_table num_rows =
+    v = create_vectors num_rows
+    x = v.at 0
+    y = v.at 1
+    u = v.at 2
+    z = v.at 3
+
+    t = Table.new [["X", x], ["Y", y], ["U", u], ["Z", z]]
+
+    Runtime.assert ((t.at "X" . value_type) == Value_Type.Integer)
+    Runtime.assert ((t.at "Y" . value_type) == Value_Type.Integer)
+    Runtime.assert ((t.at "U" . value_type) == Value_Type.Integer)
+    Runtime.assert ((t.at "Z" . value_type) == Value_Type.Integer)
     t
 
+foreign arrow int64_new = """
+    new[Int64]
+
+foreign arrow int64_plus = """
+    +[Int64]
 
 type Data
-    Value ~table
+    private Value ~table ~arrow
 
-    create num_rows = Data.Value (create_table num_rows)
+    arrow_plus self = self.arrow.at 0
+    arrow_x self = self.arrow.at 1
+    arrow_y self = self.arrow.at 2
+    arrow_u self = self.arrow.at 3
+    arrow_z self = self.arrow.at 4
+
+    create num_rows = Data.Value (create_table num_rows) <|
+        column_to_arrow v:Vector -> Array =
+            builder = int64_new.new v.length
+            v.map e-> builder.append e
+            builder.build
+
+        v = create_vectors num_rows
+
+        x = column_to_arrow (v.at 0)
+        y = column_to_arrow (v.at 1)
+        u = column_to_arrow (v.at 2)
+        z = column_to_arrow (v.at 3)
+        [int64_plus, x, y, u, z]
```

Comment:
nitpick but it is weird to read where `create_table` is separate function and the other one is not. Maybe we could just make this:

```suggestion
    create num_rows = Data.Value (create_table num_rows) (create_arrow_columns num_rows)
    
    create_arrow_columns num_rows =
        column_to_arrow v:Vector -> Array =
            builder = int64_new.new v.length
            v.map e-> builder.append e
            builder.build

        v = create_vectors num_rows

        x = column_to_arrow (v.at 0)
        y = column_to_arrow (v.at 1)
        u = column_to_arrow (v.at 2)
        z = column_to_arrow (v.at 3)
        [int64_plus, x, y, u, z]
```

Reply:
The name `create_arrow_columns` could not be found - https://github.com/enso-org/enso/actions/runs/9461815319/job/26063401099?pr=10150#step:7:1631

Reply:
Corrected in 807fd6d

Reply:
Uh sorry, indeed I forgot that in this format it should have been `Data.create_arrow_columns` in the callsite

Reply:
Or function moved to top-level like you did, also works